src/raftstore/store/worker/snap.rs

```diff
@@ -147,13 +208,15 @@ impl<T: MsgSender> Runner<T> {
         if !snap_file.exists() {
             return Err(box_err!("missing snap file {}", snap_file.path().display()));
         }
+        try!(check_abort(&abort));
         box_try!(snap_file.validate());
         let mut reader = box_try!(File::open(snap_file.path()));
 
         let timer = Instant::now();
         // Write the snapshot into the region.
         loop {
             // TODO: avoid too many allocation
+            try!(check_abort(&abort));
```

Comment:
Its goal is to abort as soon as possible rather than to avoid possible time consuming operation.